### ISSUE TITLE
Use fn keys as standard function keys only for external keyboard

### ISSUE THREAD

**Issue by tenhobi** [pqrs-org/Karabiner-Elements#3332]
Hi, is it possible to only allow f1, f2, ... keys as standard fn keys only for external keyboard?

My use case: I have K860 and it works all good by itself. But I have paragraph key where the tilda should be, so I need this software to remap that.

![image](https://user-images.githubusercontent.com/5287596/207063079-3196dbb7-c16a-4d3e-ab9f-b393890ea335.png)

This K860 keyboard also has a fn lock key, that switches between fn keys and media keys. Without Karabiner, it works just fine. At any point, I can use media keys on my MacBook, and depending on fn lock key on K860 I can switch fn/media keys.

But after using Karabiner, this functionality stop working and no matter if I lock or unlock the fn keys, it always works like media keys. For that to work, I have to select "Use all F1, F2, ... as standard function keys". Then the lock/unlock works. 

![image](https://user-images.githubusercontent.com/5287596/207063524-07ba4052-4947-4300-9a46-53a22ab94631.png)

But with that enabled, and while trying to type directly on MacBook, the f1, f2, ... keys might work as fn keys, not media keys  (depending on whether I left it lock on unlock on the external keyboard). 

Q: Is there a way how to set up "Use all F1, F2, ... as standard function keys" only for the external keyboard? Or, how to fix it other way?

**Comment by olliepop777** [comment]
I would love this feature too but I don't think it's possible in the current version. As a workaround could you remap the function keys just on the Mac's internal keyboard to the media keys?

**Comment by ivandaniel15** [comment]
I have exactly the same problem and same keyboard as well. Tried the suggestion but it doesn't work, the Mac keys keep being recognized as standard function keys. 
Also, the most annoying part is that, when karabiner is running, when I press (on the mac keyboard) the Fn Key + any F key, it instantly opens the emoji selector. If karabiner is closed, the emoji selector only comes if the Fn key is pressed and released right away, not hold. Anyone knows how to change this behaviour?

**Comment by krzysztofsroga** [comment]
The easiest way seems to disable "Use all F1, F2..." option and then use 'inherit' on internal keyboard and select F1, F2 on external non-apple keyboards

<img width="577" alt="obraz" src="https://user-images.githubusercontent.com/9088882/220901364-efb0a21e-5740-4484-a9c5-435601366772.png">

**Comment by DamianPereira** [comment]
> The easiest way seems to disable "Use all F1, F2..." option and then use 'inherit' on internal keyboard and select F1, F2 on external non-apple keyboards
> 
> <img alt="obraz" width="577" src="https://user-images.githubusercontent.com/9088882/220901364-efb0a21e-5740-4484-a9c5-435601366772.png">

That sort of works, but know pressing fn + f keys does not trigger the corresponding fn.

If you want f1 on the mac builtin keyboard to decrease brightness, f1 on an external keyboard to just be f1, but fn + f1 on external keyboard to decrease brightness, it is currently not possible with any workaround as far as I could see.

**Comment by tuckerchapin** [comment]
> If you want f1 on the mac builtin keyboard to decrease brightness, f1 on an external keyboard to just be f1, but fn + f1 on external keyboard to decrease brightness, it is currently not possible with any workaround as far as I could see.

`fn` doesn't really have a proper key code. The issue is that the internal keyboard (and maybe even external Apple keyboards?) sends `fn` events. On most keyboards, it's internal and not treated like other modifiers. So if you have a 60% keyboard where the `Esc` key also doubles as `` `~ `` on the function layer, you would physically press `fn` and `Esc` but the keyboard firmware would simply send `` ` `` which is all Karabiner would see. Whereas the internal MacBook keyboard actually sends `fn + Esc` (obviously this is a simplification since it's actually sending `key_down` and `key_up`s for each, but similar effect).

If you're using a QMK or similarly customizable firmware, you could possibly have your `fn` key fire that event as well though it's decidedly non-standard.

**Comment by damicore** [comment]
> > If you want f1 on the mac builtin keyboard to decrease brightness, f1 on an external keyboard to just be f1, but fn + f1 on external keyboard to decrease brightness, it is currently not possible with any workaround as far as I could see.
> 
> `fn` doesn't really have a proper key code. The issue is that the internal keyboard (and maybe even external Apple keyboards?) sends `fn` events. On most keyboards, it's internal and not treated like other modifiers. So if you have a 60% keyboard where the `Esc` key also doubles as `` `~ `` on the function layer, you would physically press `fn` and `Esc` but the keyboard firmware would simply send `` ` `` which is all Karabiner would see. Whereas the internal MacBook keyboard actually sends `fn + Esc` (obviously this is a simplification since it's actually sending `key_down` and `key_up`s for each, but similar effect).
> 
> If you're using QMK or similarly customizable firmware, you could possibly have your `mo(*)` bind (analogous to `fn`) fire that event as well though it's decidedly non-standard.

I would agree with you, but it somehow works like it should on MacOS Sequoia once you close karabiner. 
At least that's the case on my logitech k860 connected to my macbook air m2. It just works out of the case, but I have to use karabiner for a couple of modifications like the one in the OP.

**Comment by tenhobi** [comment]
Still persists.

**Comment by tekezo** [comment]
Please try this setting:
https://karabiner-elements.pqrs.org/docs/json/expert-complex-modifications-examples/swap-function-keys-on-external-keyboards/

**Comment by tenhobi** [comment]
Thank you. I will try in 2-3 weeks, I am out of country now. Then I will let you know
